google/cloud/bigtable/tests/instance_admin_emulator.cc

```diff
@@ -187,17 +195,21 @@ class InstanceAdminEmulator final
   grpc::Status ListClusters(grpc::ServerContext* context,
                             btadmin::ListClustersRequest const* request,
                             btadmin::ListClustersResponse* response) override {
-    std::string prefix = request->parent();
+    std::string instance_path =
+        request->parent().substr(0, request->parent().find_last_of("/"));
+    std::string prefix = request->parent() + "/clusters/";
 
-    // magical instance name "-" must return all clusters in the project
-    if (request->parent().back() == '-') {
-      for (auto const& kv : clusters_) {
-        *response->add_clusters() = kv.second;
+    for (auto const& instance : instances_) {
+      if (std::string::npos == instance.first.find(instance_path)) {
+        continue;
       }
-    } else {
-      for (auto const& kv : clusters_) {
-        if (kv.first.find(prefix) != std::string::npos) {
-          *response->add_clusters() = kv.second;
+      for (auto const& cluster : clusters_) {
+        if (request->parent().back() == '-') {
```

Comment:
Same here, explaining why only checking the last character is enough would be useful.

Reply:
And actually this code works incorrectly for a request with 'foo-' as the instance id.  That is an invalid instance id, but it should not return all the clusters either.